spec/unit/defaults_spec.rb

```diff
@@ -57,6 +57,20 @@ module Vault
       end
     end
 
+    describe ".namespace" do
```

Comment:
Isn't the desired behavior to use the env var if present, and to be nil if not, such that the header isn't added at all?